report/2.Analysis/ArduinoLanguagesCtrl.tex

```diff
@@ -1,70 +1,130 @@
+Similarly, functions aren't given bodies in the class declaration either.
+Instead, the functions must be given a body outside of the class, and the variables must be initialized in the constructor function, also outside of the body.
+
+\textit{Preprocessor directives} such as including a header file, or setting up macros are written with a pound symbol (snippet \ref{lis:arduinoExample}, lines 6-9).
+
+\textit{Comments} can be written in two different ways.
+The end-of-line comment is prefixed with two slashes (snippet \ref{lis:arduinoExample}, line 6).
+The multi-line comment is prefixed with a slash followed by an asterisk (\texttt{"/*"}), and postfixed with an asterisk followed by a slash (\texttt{"*/"}) (snippet \ref{lis:arduinoExample}, line 1-4).
 
 \begin{lstlisting}[language=C++,label=lis:arduinoExample,caption=An example program written in the Arduino language.,firstnumber=1]
 /*
- * Preprocessor directives
+ * This program makes an LED connected to pin 13 on the arduino
+ * board blink on and off every 500 milliseconds
  */
 
-const int PIN13 = 13;
+#include "Arduino.h" //The Arduino standard library
+#include "Blinker.h" //The Blinker custom library
+
+#define LED_PIN 13 //Symbolic constant
+
+const int BLINK_DELAY = 250; //Constant, global integer variable
+
+Blinker blinker(LED_PIN, BLINK_DELAY); //Object instantiation
+
+void setup() {} //No setup code for this program
+
+void loop()
+{
+    blinker.Blink(); //Program blinks infinitely
+}
 
-void setup() {
-    // Setup code
-    pinMode(PIN13, OUTPUT); // Set digital pin 13 to output
+Blinker::Blinker(int pin, int msDelay)
+{
+	pinMode(pin, OUTPUT);
+	_pin = pin;
+	_msDelay = msDelay;
 }
 
-void loop() {
-    // Loop code is looped infinitely
-    digitalWrite(PIN13, HIGH); // Set output for pin 13 to high
-    delay(500);                // Delay for 500 milliseconds
-    digitalWrite(PIN13, LOW);  // Set output for pin 13 to low
-    delay(500);                // Delay for 500 milliseconds
+void Blinker::Blink()
+{
+	digitalWrite(_pin, HIGH);
+	delay(_msDelay);
+	digitalWrite(_pin, LOW);
+	delay(_msDelay);
 }
 
 \end{lstlisting}
 
+\begin{lstlisting}[language=C++,label=lis:arduinoHeader,caption=The Blinker.h header file.,firstnumber=1]
+class Blinker
+{
+	public:
+		Blinker(int pin, int msDelay); //Constructor
+		void Blink();
+	private:
+		int _pin;
+		int _msDelay;
+};
+\end{lstlisting}
+
 % Source: concurrency.cc
 \textbf{Occam-pi} is a language that focuses on making parallel programming easy for programmers.
 It is based on Occam that first appeared in 1983.
-The full documentation of Occam-pi can be read on the official website of the language, \url{concurrency.cc}, but here is a short description of some of the syntax.
+The full documentation of Occam-pi can be read on the official website of the language, \url{concurrency.cc}, but following here is a short description of some of the syntax, and an example program in Snippet \ref{lis:occamExample}.
```

Comment:
Remove 'here'. 
better with "following is a short..." rather than "following here is a short"

Reply:
Maybe remove , but? Make it a seperate new sentence

Reply:
^YES